### PR TITLE
AVM: base64_decode can decode padded or unpadded encodings

### PR DESCRIPTION
**Pull request by jannotti**
base64 decoding now operates on padded or unpadded encodings.  Since \r and \n are ignored while decoding, the approach we take is to look for padding, and use the padded encoding if we find any, or raw if we don't. (This contrasts with how we (I?) expected to do it - by checking if the padding was correct and stripping it off.  But that would require scanning the string, not just checking the length.)

### CODE REVIEW

File: data/transactions/logic/eval_test.go [hunk 1]
```diff
@@ -4649,12 +4649,12 @@ By Herman Melville`, "",
 		{"cGFk=", "StdEncoding", "pad", "input byte 4"},
 		{"cGFk==", "StdEncoding", "pad", "input byte 4"},
 		{"cGFk===", "StdEncoding", "pad", "input byte 4"},
-		// Ensures that even correct padding is illegal if not needed
+		// Ensures that extra padding, even if 0%4
 		{"cGFk====", "StdEncoding", "pad", "input byte 4"},
 
-		// Test that padding must be present to make len = 0 mod 4.
+		// Test that padding must be correct of absent
```
Comment (tzaffi) [inline data/transactions/logic/eval_test.go:4655]:
```suggestion
		// Test that padding must be correct if absent
```

Comment (jannotti) [reply data/transactions/logic/eval_test.go:4655]: ha, single character typos can be confusing.  I meant "or".

**Review by tzaffi — APPROVED** [review]
LGTM - left an optional suggestion to fix a typo in a comment

**Review by tzaffi — APPROVED** [review]
LGTM

**Pull request merged**